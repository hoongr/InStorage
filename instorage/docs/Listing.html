<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: Listing</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: Listing</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>Listing<span class="signature">(address, available, dateEndAvailable, dateStartAvailable, hostUserID, listingID, photoURL, rentalPrice, squareFootageEstimate, coordinates, distance)</span><span class="type-signature"></span></h2>
        
            <div class="class-description">Holds attributes of a apartment listing.</div>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    
    <h2>Constructor</h2>
    

    
    <h4 class="name" id="Listing"><span class="type-signature"></span>new Listing<span class="signature">(address, available, dateEndAvailable, dateStartAvailable, hostUserID, listingID, photoURL, rentalPrice, squareFootageEstimate, coordinates, distance)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    Takes in a JSON from Firebase containing listing attributes and creates an object from it.
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>address</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">listing address</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>available</code></td>
            

            <td class="type">
            
                
<span class="param-type">boolean</span>


            
            </td>

            

            

            <td class="description last">if listing is available</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>dateEndAvailable</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">end date of availability</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>dateStartAvailable</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">start date of availability</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>hostUserID</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">ID of host of listing</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>listingID</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">ID of listing</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>photoURL</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">URL for listing photo</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>rentalPrice</code></td>
            

            <td class="type">
            
                
<span class="param-type">float</span>


            
            </td>

            

            

            <td class="description last">price of listing</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>squareFootageEstimate</code></td>
            

            <td class="type">
            
                
<span class="param-type">int</span>


            
            </td>

            

            

            <td class="description last">estimate of listing square footage</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>coordinates</code></td>
            

            <td class="type">
            
                
<span class="param-type">Array.&lt;int></span>


            
            </td>

            

            

            <td class="description last">of the apartment</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>distance</code></td>
            

            <td class="type">
            
                
<span class="param-type">float</span>


            
            </td>

            

            

            <td class="description last">miles between listing and current coordinates</td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="models_listing.js.html">models/listing.js</a>, <a href="models_listing.js.html#line19">line 19</a>
    </li></ul></dd>
    

    

    

    
</dl>




















    
    </div>

    

    

    

    

    

    

    
        <h3 class="subsection-title">Members</h3>

        
            
<h4 class="name" id="address"><span class="type-signature"></span>address<span class="type-signature"></span></h4>










    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>address</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">listing address</td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="models_listing.js.html">models/listing.js</a>, <a href="models_listing.js.html#line35">line 35</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="available"><span class="type-signature"></span>available<span class="type-signature"></span></h4>










    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>available</code></td>
            

            <td class="type">
            
                
<span class="param-type">boolean</span>


            
            </td>

            

            

            <td class="description last">is listing available</td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="models_listing.js.html">models/listing.js</a>, <a href="models_listing.js.html#line44">line 44</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="coordinates"><span class="type-signature"></span>coordinates<span class="type-signature"></span></h4>










    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>Listing.coordinates</code></td>
            

            <td class="type">
            
                
<span class="param-type">Array.&lt;int></span>


            
            </td>

            

            

            <td class="description last">coordinates of the listed apartment</td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="models_listing.js.html">models/listing.js</a>, <a href="models_listing.js.html#line116">line 116</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="dateEndAvailable"><span class="type-signature"></span>dateEndAvailable<span class="type-signature"></span></h4>










    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>Listing.dateEndAvailable</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">listing end date</td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="models_listing.js.html">models/listing.js</a>, <a href="models_listing.js.html#line53">line 53</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="dateStartAvailable"><span class="type-signature"></span>dateStartAvailable<span class="type-signature"></span></h4>










    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>Listing.dateStartAvailable</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">listing start date</td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="models_listing.js.html">models/listing.js</a>, <a href="models_listing.js.html#line62">line 62</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="distance"><span class="type-signature"></span>distance<span class="type-signature"></span></h4>










    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>Listing.distance</code></td>
            

            <td class="type">
            
                
<span class="param-type">int</span>


            
            </td>

            

            

            <td class="description last">distance between user and listed apartment</td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="models_listing.js.html">models/listing.js</a>, <a href="models_listing.js.html#line125">line 125</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="hostUserID"><span class="type-signature"></span>hostUserID<span class="type-signature"></span></h4>










    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>Listing.hostUserID</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">listing's host user ID</td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="models_listing.js.html">models/listing.js</a>, <a href="models_listing.js.html#line71">line 71</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="listingID"><span class="type-signature"></span>listingID<span class="type-signature"></span></h4>










    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>Listing.listingID</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">listingID of listing</td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="models_listing.js.html">models/listing.js</a>, <a href="models_listing.js.html#line80">line 80</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="photoURL"><span class="type-signature"></span>photoURL<span class="type-signature"></span></h4>










    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>Listing.photoURL</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last">listing's photo url</td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="models_listing.js.html">models/listing.js</a>, <a href="models_listing.js.html#line89">line 89</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="rentalPrice"><span class="type-signature"></span>rentalPrice<span class="type-signature"></span></h4>










    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>Listing.rentalPrice</code></td>
            

            <td class="type">
            
                
<span class="param-type">float</span>


            
            </td>

            

            

            <td class="description last">listing rental price</td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="models_listing.js.html">models/listing.js</a>, <a href="models_listing.js.html#line98">line 98</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="squareFootageEstimate"><span class="type-signature"></span>squareFootageEstimate<span class="type-signature"></span></h4>










    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>Listing.squareFootageEstimate</code></td>
            

            <td class="type">
            
                
<span class="param-type">int</span>


            
            </td>

            

            

            <td class="description last">square footage estimate of listing</td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="models_listing.js.html">models/listing.js</a>, <a href="models_listing.js.html#line107">line 107</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
    

    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="BookingRequest.html">BookingRequest</a></li><li><a href="EmailManager.html">EmailManager</a></li><li><a href="FirebaseManager.html">FirebaseManager</a></li><li><a href="Listing.html">Listing</a></li><li><a href="MovingJob.html">MovingJob</a></li><li><a href="UserProfile.html">UserProfile</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 3.6.3</a> on Sat Nov 30 2019 19:48:32 GMT-0800 (Pacific Standard Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>