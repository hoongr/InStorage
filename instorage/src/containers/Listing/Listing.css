@import url('https://fonts.googleapis.com/css?family=Montserrat&display=swap');
@import url('https://fonts.googleapis.com/css?family=Lato&display=swap');

.pad-container {
	margin-top: 30px;
}

.custom-text {
	font-size: 1em;
	margin-top: 1.33em;
	margin-bottom: 1.33em;
	margin-left: 0;
	margin-right: 0;
	font-weight: bold;
	padding-right: 7px;
}

.btn-custom {
	background-color: #274060;
	color: white;
	width: 240px;
	margin-left: auto;
	margin-right: auto;
	margin-top: 20px;
	display: block;
}

.btn:hover {
	color: #fff !important;
	text-decoration: none;
}

.w-custom {
	width: 100px;
	display: block;
	margin-left: auto;
	margin-right: auto;
	margin-bottom: 20px;
}

.profile-pic {
	width: 50%;
	height: 80%;
	display: block;
	margin-left: auto;
	margin-right: auto;
	padding-top: 20px;
}

.name-text {
	display: block;
	margin-left: auto;
	margin-right: auto;
	font-size: 24px;
}

.apt-info{
	width: 40%;
	margin-right: 4%;
}

.range-pick{
	width: 80%;
}

.booking-form{
	padding-top: 10px;
	width: 350px;
	height: 540px;
	border: 1px solid #E1E1E1;
	background: white;
	z-index: 5 !important;
	/*position: relative;*/
}

div.DateRangePicker{
	margin-top: 0px;
	margin-left: 0px;
	margin-right: 0px;
}

div.DateInput{
	width: 100px;
}

#your_unique_start_date_id.DateInput_input{
	font-size:15px;
	font-family: Helvetica;
}

#your_unique_end_date_id.DateInput_input{
	font-size:15px;
	font-family: Helvetica;
}

.mx25 {
	margin-left: 25px;
	margin-right: 25px;
}

.my1rem {
	margin-top: 1rem;
	margin-bottom: 1rem;
}

.mb1rem {
	margin-bottom: 1rem !important;
}

.form-check {
	margin-left: 25px;
}

.form-check-input {
	margin-left: 0.1px;
	/*padding-left: 13px;*/
}

.form-check-label{
	padding-left: 20px;
}

.actual-content{
	/* font-family: 'Lato', sans-serif; */
	padding-top: 35px;
}

.address{
	/* font-family: 'Montserrat', sans-serif; */
	font-size: 25px;
	font-weight: 700;
	margin-bottom: 20px;
	padding-right: 5px;
}

.apt-pic {
	object-fit: cover;
	width: 100%;
	max-height: 65vh;
}

.my-profile-pic {
	height: 80px;
	width: 80px;
	display: block;
	margin-left: auto;
	margin-right: auto;
}

.my-listing-date .DateInput_input{
	margin-top: 0px !important;
}

.my-listing-date .DateRangePickerInput__withBorder{
	width: 298px;
}

.my-listing-date div.DateInput.DateInput_1{
	width: 128px;
}

.my-listing-date #your_unique_start_date_id.DateInput_input{
	font-size:15px;
	text-align: center;
	font-weight: 200;
	/* font-family: 'Lato', sans-serif; */
}

.my-listing-date #your_unique_end_date_id.DateInput_input{
	font-size:15px;
	font-weight: 200;
	text-align: center;
	/* font-family: 'Lato', sans-serif; */
}

.my-listing-date .DateRangePicker {
	/* margin: 0 auto; */

}

label.form-label{
	margin-bottom: 0px;
}

.form-group{
	margin-bottom: 5px;
}

.my-location-search-input .location-search-input{
	width: 298px;
}

.ant-time-picker-panel-select li{
	display: block !important;
}
