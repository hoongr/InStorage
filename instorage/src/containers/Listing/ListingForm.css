
.title {
	text-align: center;
	padding-top: 40px;
}

.card-margin {
	margin: 40px;
}

.form-padding {
	padding: 10px;
}

.btn-custom {
	background-color: #274060;
	color: white;
}

.btn:hover {
	color: #fff !important;
	text-decoration: none;
}

