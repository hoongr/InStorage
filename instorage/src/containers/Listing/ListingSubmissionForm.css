.my-submission-form-date #start_date_id.DateInput_input{
    font-size:15px;
    font-family: Helvetica;
}

.my-submission-form-date #end_date_id.DateInput_input{
    font-size:15px;
    font-family: Helvetica;
}

.my-submission-form-location .location-search-input{
    width: 100%;
}

.card-container {
	margin: 0 auto;
	float: none;
	margin-bottom: 10px;
	width: 100%;
	max-width: 700px;
}
