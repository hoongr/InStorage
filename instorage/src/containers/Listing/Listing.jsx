import React from "react";
import "./Listing.css";
import { Alert, Button } from "react-bootstrap";
import { Form, Col, Card, Row, Image } from "react-bootstrap";
import "react-dates/initialize";
import { DateRangePicker } from "react-dates";
import "react-dates/lib/css/_datepicker.css";
import { TimePicker } from "antd";
import "antd/dist/antd.css";
import FirebaseManager from "../../api/routes/firebaseManager";
import "firebase/storage";
import "../../models/listing.js";
import { withAuthorization } from "../../Session";
import LocationSearchInput from "../Map/LocationSearchInput";
import ReactDependentScript from "react-dependent-script";

class Listing extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      listing: {},
      userId: "",
      listingId: "",
      startDate: "",
      endDate: "",
      moveTime: "",
      needMover: false,
      showAlert: false,
      alertType: "",
      bookingAlertMessage: "",
      bookingAlertTitle: "",
      hostName: "",
      clientAddress: "",
      photo: "",
      photoKey: "",
      timeKey: "",
      addressKey: "",
      coordinates: "",
      inventoryString: "",
      profilePicture: "",
      hostAddress: "",
    };
  }

  componentDidMount() {
    const { auth } = this.props.firebase;
    console.log(auth.currentUser.uid);
    this.setState({ userId: auth.currentUser.uid });

    this.getListing(this.props.match.params.listingId);
  }

  change = e => {
    const target = e.target;
    const value = target.type === "file" ? target.files[0] : target.value;
    const name = target.name;
    console.log(name+" "+value)
    this.setState({
      [name]: value
    });
  };

  getUser(userId) {
    console.log("looking for user ", userId);
    FirebaseManager.getUserProfile(userId).then(user =>
      this.setState({
        hostName: user._name,
        profilePicture: user._profilePicURL
      })
    );
  }

  getListing(listingId) {
    const listing = FirebaseManager.getListing(listingId).then(
      listing => {
        console.log("yes, ", listing);
        this.setState({
          listing: listing
        });

        this.getUser(listing._hostUserID);
        this.state.hostAddress = this.state.listing._address.split(',');
        console.log(this.state.listing);
      }
    );
  }

  functionThatResetsTheFileInput() {
    let randomString = Math.random().toString(36);

    this.setState({
      photoKey: randomString,
      addressKey: randomString,
      timeKey: randomString
    });
  }

  onTimeChange = (time, timeString) => {
    this.setState({
      moveTime: timeString
    });
  };

  parseInventoryList = (inventoryString) => {
    var inventoryArr = inventoryString.split(',');
    return inventoryArr;
  }

  requestToBook = e => {
    e.preventDefault();
    const { photo } = this.state;
    const { storage } = this.props.firebase;
    const randomNum = Math.floor(Math.random() * 100000000000000);
    const uploadTask = storage
      .ref(`inventory/${randomNum}${photo.name}`)
      .put(photo, { contentType: photo.type });
    uploadTask.on(
      "state_changed",
      snapshot => {
        console.log("uploading");
      },
      error => {
        console.log(error);
        this.setState({
          showAlert: true,
          alertType: "danger",
          bookingAlertMessage: "Try again!",
          bookingAlertTitle: "Something went wrong",
          endDate: "",
          startDate: "",
          needMover: false,
          clientAddress: "",
          photoURL: "",
          inventoryString: "",
          startDateMoment: null,
          endDateMoment: null
        });
        this.functionThatResetsTheFileInput();
      },
      () => {
        this.setState({
          photoURL: `https://firebasestorage.googleapis.com/v0/b/instorage-6b1ea.appspot.com/o/inventory%2F${randomNum}${photo.name}?alt=media`
        });
        console.log(this.state.photoURL);
        const bookingRequest = {
          clientUserID: this.state.userId,
          dateEnd: this.state.endDate,
          dateStart: this.state.startDate,
          moveTime: this.state.moveTime,
          hostUserID: this.state.listing._hostUserID,
          listingID: this.props.match.params.listingId,
          needMover: this.state.needMover.toString(),
          requestState: "pending",
          clientAddress: this.state.clientAddress,
          hostAddress: this.state.listing._address,
          inventoryList: this.parseInventoryList(this.state.inventoryString),
          inventoryPhotoURL: this.state.photoURL
        }
        FirebaseManager.addBookingRequest(
          bookingRequest
        );
        this.setState({
          showAlert: true,
          alertType: "success",
          bookingAlertMessage:
            "You will receive an email when the host accepts or rejects your request. Sit tight!",
          bookingAlertTitle: "Request received",
          endDate: "",
          startDate: "",
          moveTime: "",
          needMover: false,
          clientAddress: "",
          coordinates: "",
          fullAddressInfo: "",
          photoURL: "",
          inventoryString: "",
          startDateMoment: null,
          endDateMoment: null
        });
        this.functionThatResetsTheFileInput();
      }
    );
  };

  render() {
    return (
      <ReactDependentScript
        loadingComponent={<div>map loading...</div>}
        scripts={[
          "https://maps.googleapis.com/maps/api/js?key=AIzaSyAjOpk-lRnAG_ItuCpJDU8Bo5lgqUKhSr8&libraries=places"
        ]}
      >
        <div className="pb-4">
          <Alert
            show={this.state.showAlert}
            variant={this.state.alertType}
            className="alert-centered"
          >
            <Alert.Heading>{this.state.bookingAlertTitle}</Alert.Heading>
            <p>{this.state.bookingAlertMessage}</p>
            <hr />
            <div className="d-flex justify-content-end">
              <Button
                onClick={() => {
                  this.setState({ showAlert: false });
                  this.props.history.push('/home');
                  // window.location.reload();
                }}
                variant="outline-dark"
              >
                OK
              </Button>
            </div>
          </Alert>
          <div className="pic-wrapper">
            <img
              className="apt-pic"
              src={this.state.listing._photoURL}
              alt=""
            />
          </div>

          <div className="d-flex flex-row justify-content-center form-padding actual-content">
            <div className="apt-info">
              <div
                className="d-flex justify-between border-bottom border-dark col"
                style={{ padding: "10px" }}
              >
                <Col sm={8}>
                  <h2>
                    {this.state.hostAddress[0]}<br/>
                    {this.state.hostAddress[1]}<br/>
                    {this.state.hostAddress[2]}, {this.state.hostAddress[3]}
                  </h2>
                </Col>
                <Col sm={4}>
                  <Image
                    className="rounded-circle my-profile-pic"
                    style={{
                      objectFit: "cover",
                    }}
                    src={this.state.profilePicture}
                    alt=""
                    onClick={() => {
                      this.props.history.push("/user/" + this.state.listing._hostUserID);
                    }}
                  />
                  <h6 className="text-center pt-2">{this.state.hostName}</h6>
                </Col>
              </div>
              <div className="col" style={{ padding: "10px" }}>
                  <p>
                    <span className="custom-text">
                      Available Dates:
                    </span>
                    <span>
                      {this.state.listing._dateStartAvailable}
                      <span className="custom-text pl-1">
                        {" "}to
                      </span>
                      {this.state.listing._dateEndAvailable}
                    </span> 
                  </p>
                  <p>
                    <span className="custom-text">
                      Size of Space:
                    </span>
                    <span>
                      {this.state.listing._squareFootageEstimate} square feet
                    </span> 
                  </p>
                  <p>
                    <span className="custom-text">
                      Rental Price:
                    </span>
                    <span>
                      ${this.state.listing._rentalPrice} per week
                    </span> 
                  </p>
                  <button 
                    className="btn btn-sm"
                    style={{
                      background: "#7CC6FE",
                      color: "white",
                      width: "120px",
                    }} 
                    onClick={() => {
                      this.props.history.push("/user/" + this.state.listing._hostUserID);
                    }}
                  >
                    Contact Host
                  </button>
              </div>
            </div>

            <Form className="booking-form">
              <Form.Group className="mx25 mb1rem my-listing-date">
                <Form.Label>Enter your rental period</Form.Label>
                <DateRangePicker
                  startDate={this.state.startDateMoment}
                  startDateId="your_unique_start_date_id"
                  endDate={this.state.endDateMoment}
                  endDateId="your_unique_end_date_id"
                  isOutsideRange={(day) => {
                    if (this.state.focusedInput == "endDate") {
                      return day.isAfter(this.state.listing._dateEndAvailable) || day.isBefore(this.state.listing._dateStartAvailable);
                    }
                    if (this.state.focusedInput == "startDate") {
                      return day.isBefore(this.state.listing._dateStartAvailable) || day.isAfter(this.state.listing._dateEndAvailable);
                    }
                  }}
                  onDatesChange={({ startDate, endDate }) => {
                    this.setState({
                      startDateMoment: startDate,
                      endDateMoment: endDate,
                      startDate:
                        startDate === null
                          ? null
                          : startDate.toISOString().substr(0, 10),
                      endDate:
                        endDate === null
                          ? null
                          : endDate.toISOString().substr(0, 10)
                    });
                  }}
                  focusedInput={this.state.focusedInput}
                  onFocusChange={focusedInput =>
                    this.setState({ focusedInput })
                  }
                />
              </Form.Group>
              <Form.Group className="mb1rem mx25 pick-up-time-input">
                <Form.Label>Enter your drop off time</Form.Label>
                <br />
                <TimePicker
                  onChange={this.onTimeChange}
                  use12Hours
                  size="large"
                  format="h:mm A"
                  className="time-picker"
                />
              </Form.Group>

              <Form.Group className="mx25 mb1rem my-location-search-input">
                <Form.Label>Enter your location</Form.Label>
                <br />
                <LocationSearchInput
                  placeHolder={"Enter your location"}
                  value={this.state.clientAddress}
                  onChange={(address, coordinates, fullInfo) =>
                    this.setState({
                      clientAddress: address,
                      coordinates: coordinates,
                      fullAddressInfo: fullInfo
                    })
                  }
                ></LocationSearchInput>
              </Form.Group>

              <Form.Group className="mx25 mb1rem" controlId="formBasicCheckbox">
                <input
                  className="form-check-input"
                  type="checkbox"
                  value=""
                  id="defaultCheck1"
                  checked={this.state.needMover}
                  onChange={() => {
                    this.setState({ needMover: !this.state.needMover });
                  }}
                />
                <label className="form-check-label" htmlFor="defaultCheck1">
                  Need a mover? Costs $5/mile
                </label>
              </Form.Group>
              <Form.Group controlId="formInventoryList" className="mb1rem mx25">
                <Form.Label>List your inventory, separated by commas</Form.Label>
                <Form.Control
                  type="text"
                  name="inventoryString"
                  onChange={e => this.change(e)}/>
              </Form.Group>
              <Form.Group className="mb1rem mx25" controlId="photo">
                <Form.Label>Upload a photo of your packed items</Form.Label>
                <Form.Control
                  type="file"
                  name="photo"
                  key={this.state.photoKey || ""}
                  onChange={e => this.change(e)}
                />
              </Form.Group>
              <button
                type="button"
                className="btn btn-custom btn-lg mb1rem"
                onClick={e => this.requestToBook(e)}
              >
                Request to Book
              </button>
            </Form>
          </div>
        </div>
      </ReactDependentScript>
    );
  }
}

const condition = authUser => !!authUser;

export default withAuthorization(condition)(Listing);
