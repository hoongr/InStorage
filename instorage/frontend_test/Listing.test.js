import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

import Listing from "../src/containers/Listing/Listing"
import ListingSubmissionForm from "../src/containers/Listing/ListingSubmissionForm"

describe('Listing', () => {
	const wrapper = shallow(<Listing />);

	test("renders without crashing", () => {
		expect(wrapper).toBeDefined();
	});

	test("should match the snapshot", () => {
	  expect(wrapper).toMatchSnapshot();
	});
})

describe('ListingSubmission', () => {
	const wrapper = shallow(<ListingSubmissionForm />);

	test("renders without crashing", () => {
		expect(wrapper).toBeDefined();
	});

	test("should match the snapshot", () => {
	  expect(wrapper).toMatchSnapshot();
	});
})