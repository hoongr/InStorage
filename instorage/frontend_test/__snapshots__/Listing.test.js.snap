// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Listing should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`ListingSubmission should match the snapshot 1`] = `ShallowWrapper {}`;
